proxy/src/control/observe.rs

```diff
@@ -24,7 +24,7 @@ pub struct Observe {
 pub struct TapEvents {
     rx: futures_mpsc_lossy::Receiver<Event>,
     remaining: usize,
-    current: IndexMap<Arc<ctx::http::Request>, ()>,
+    current: IndexMap<ctx::http::RequestId, Arc<ctx::http::Request>>,
```

Comment:
I think it might be more efficient to avoid storing the request ID twice, once as the key, and once as a field inside the value. I thought about creating a `RequestById(ctx::http::RequestId)` type that manually forwards `Hash`, `Eq`, `PartialEq`, to `self.0`, but it seems like that's a lot of boilerplate and there must be some way to avoid that boilerplate. Suggestions?